server/sonar-server/src/main/java/org/sonar/server/measure/persistence/MeasureDao.java

```diff
@@ -53,6 +55,18 @@ public MeasureDto findByComponentKeyAndMetricKey(DbSession session, String compo
     });
   }
 
+  public List<PastMeasureDto> selectByComponentUuidAndProjectSnapshotIdAndMetricIds(final DbSession session, final String componentUuid,
+    final long projectSnapshotId,
+    List<Integer> metricIds) {
```

Comment:
should you take a Set rather than a list?
you don't care about the order of ids but you really need to not request twice (or more) for a specified metric id, hence a Set